tools/topology/sof-imx8mp-keyword-detect.m4

```diff
@@ -0,0 +1,96 @@
+#
+# Topology for i.MX8 with wm8960 codec for
+# keyword detection and triggering use case.
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+include(`sai.m4')
+# Include i.MX8 DSP configuration
+include(`platform/imx/imx8qxp.m4')
+
+define(KWD_PIPE_SCH_DEADLINE_US, 20000)
+
+define(DMIC_16k_PCM_NAME, `SAI3')
+
+# This is for iMX8M PLUS
+#
+# PCM 0 <-------+- KPBM 0 <-- B0 <-- SAI3 (wm8960)
+#               |
+# Keyword <-----+
+dnl PIPELINE_PCM_ADD(pipeline,
+dnl     pipe id, pcm, max channels, format,
+dnl     period, priority, core,
+dnl     pcm_min_rate, pcm_max_rate, pipeline_rate,
+dnl     time_domain, sched_comp)
+
+# Passthrough capture pipeline 1 on PCM 0 using max 2 channels.
+# Schedule 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-kfbm-no-lp-capture.m4,
+	1, 0, 2, s16le,
+	KWD_PIPE_SCH_DEADLINE_US, 0, 0,
+	16000, 16000, 16000)
+
+#
+# DAIs configuration
+#
+
+dnl DAI_ADD(pipeline,
+dnl     pipe id, dai type, dai_index, dai_be,
+dnl     buffer, periods, format,
+dnl     deadline, priority, core, time_domain)
+
+# This is for iMX8M PLUS
+# capture DAI is SAI3 using 2 periods
+# Buffers use s16le format, with 320 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+        1, SAI, 3, sai3-wm8960-hifi,
+        PIPELINE_SINK_1, 2, s16le,
+        KWD_PIPE_SCH_DEADLINE_US,
+        0, 0, SCHEDULE_TIME_DOMAIN_DMA)
+
+# keyword detector pipe
+dnl PIPELINE_ADD(pipeline,
+dnl     pipe id, max channels, format,
+dnl     period, priority, core,
+dnl     sched_comp, time_domain,
+dnl     pcm_min_rate, pcm_max_rate, pipeline_rate)
+PIPELINE_ADD(sof/pipe-detect.m4,
+	2, 2, s16le,
+	KWD_PIPE_SCH_DEADLINE_US, 1, 0,
+	PIPELINE_SCHED_COMP_1,
+	SCHEDULE_TIME_DOMAIN_TIMER,
```

Comment:
I thought we need SCHEDULE_TIME_DOMAIN_DMA here, right ?

Reply:
We need DMA for the pipeline that includes the DAI but timer for the second one.

Reply:
> We need DMA for the pipeline that includes the DAI but timer for the second one.

It's scheduled on PIPELINE_SCHED_COMP_1 which is SAI3 and DMA driven, isn't it ? So why TIMER ?

Reply:
@ViorelSuman We can try putting both on DMA, but I believe there were issues (in the sense that it wouldn't run) when doing that. @cristinailie Do you remember if we have actually tried that?

Reply:
I just tried with both on DMA and I cannot see any difference in behavior.